synapse/lib/json.py

```diff
@@ -0,0 +1,204 @@
+import io
+import os
+import json
+import logging
+
+from typing import Any, BinaryIO, Callable, Iterator, Optional
+
+import orjson
+
+import synapse.exc as s_exc
+
+logger = logging.getLogger(__name__)
+
+def _fallback_loads(s: str | bytes) -> Any:
+
+    try:
+        return json.loads(s)
+    except json.JSONDecodeError as exc:
+        raise s_exc.BadJsonText(mesg=exc.args[0])
+
+def loads(s: str | bytes) -> Any:
+    '''
+    Deserialize a JSON string.
+
+    Similar to the standard library json.loads().
+
+    Arguments:
+        s (str | bytes): The JSON data to be deserialized.
```

Comment:
We don't need type annotations in-line with docs if we're annotating them on the functions directly.
```suggestion
        s: The JSON data to be deserialized.
```

Reply:
We should be able to remove the doc annotations here and leave the python declarations in place.

Reply:
This seems like it's part of the discussion of SYN-8838. I'm going to leave it how it is for now.